### PR TITLE
215 Text Image Super Resolution with OpenVINO

### PR DESCRIPTION
**Pull request by FisherWY**
This is a notebook using text image super resolution model to super resolute an image with blur text.

The flowchart is below:
Flowchart

The output image is below:
Output

### CODE REVIEW

**Comment by FisherWY** [comment]
@ryanloney It seems that my understanding of what the code_check checks has gone a bit off. I thought the check was for the existence of the binder URL, so asked the mentor in the mailing list, here is the [thread](https://lists.01.org/hyperkitty/list/gsoc@lists.01.org/thread/Q66I6D5GPX3NRZ5NMUUDY7TH5GPIOONZ/). Mentor said don't worry about it, binder URL will be added after merge.
So should I now add the notebook to readme.md and notebooks/readme.md before merge?

**Comment by FisherWY** [comment]
@ryanloney I have added the notebook to readme.md and notebooks/readme.md. All checks have passed in my fork repository.

File: notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "ie = Core()\n",
+    "model = ie.read_model(model=model_xml_path)\n",
+    "compiled_model = ie.compile_model(model=model, device_name=DEVICE)\n",
+    "\n",
+    "input_layer_ir = next(iter(compiled_model.inputs))\n",
+    "output_layer_ir = next(iter(compiled_model.outputs))\n",
```
Comment (Debskij) [inline notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb:104]:
```suggestion
    "input_layer_ir = compiled_model.input(0)\n",
    "output_layer_ir = compiled_model.output(0)\n",
```

**Review by Debskij — CHANGES_REQUESTED** [review]
Thanks for your contribution @FisherWY!

**Comment by Debskij** [comment]
Please remove .bin and .xml files from model directory and use omz_downloader instead.

**Comment by FisherWY** [comment]
Model files removed and use omz_downloader instead. @Debskij

**Review by Debskij — APPROVED** [review]

File: notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+   "metadata": {
+    "collapsed": true,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "# Text Image Super Resolution with OpenVINO\n",
+    "\n",
+    "Super Resolution is the process of enhancing the quality of an image by increasing the pixel count using deep learning. This notebook shows the Text Image Super Resolution (TISR) which takes just one low resolution image with blured text, and output a high resolution image which have clear text. We use a model called [text-image-super-resolution-0001](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/intel/text-image-super-resolution-0001/README.md) which is available from the Open Model Zoo."
```
Comment (paularamo) [inline notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb:14]:
```suggestion
    "Super Resolution is the process of enhancing the quality of an image by increasing the pixel count using deep learning. This notebook shows the Text Image Super Resolution (TISR) which takes just one low resolution image with blured text, and output a high resolution image which have clear text. We use a model called [text-image-super-resolution-0001](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/intel/text-image-super-resolution-0001/README.md) which is available from the [Open Model Zoo.](https://docs.openvino.ai/latest/model_zoo.html)"
```

Comment (paularamo) [reply notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb:14]: Please include the links of the tools we are using in this notebook.

File: notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    "from notebook_utils import load_image"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare the model we need\n",
+    "\n",
+    "We use `omz_downloader` to download the model from open model zoo. The models in open model zoo are already converted to OpenVINO Intermediate Representation (IR), so there is no need to use `omz_converter`.  \n",
```
Comment (paularamo) [inline notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb:52]: Please include the link to the tools you are referred to. For example https://docs.openvino.ai/latest/omz_tools_downloader.html

File: notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Load the Text Image Super Resolution Model\n",
+    "\n",
+    "### After preparation, now we are going to load the model\n",
+    "\n",
+    "The Text Image Super Resolution model expects one input: the input shape format as (B, C, H, W), B is batch-size, C is number of channels, H is image height, W is image width. By default the input shape is (1, 1, 360, 640).\n",
```
Comment (paularamo) [inline notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb:95]:
```suggestion
    "TISR model expects one input: the input shape format as (B, C, H, W), B is batch-size, C is number of channels, H is image height, W is image width. By default the input shape is (1, 1, 360, 640).\n",
```

File: notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Load Image And Prepare For Inference\n",
+    "\n",
+    "Here we define some functions which will be used to convert image and visualization.\n",
+    "Then we load the source image, convert to gray scale and reshape it."
```
Comment (paularamo) [inline notebooks/215-text-image-superresolution/215-text-image-superresolution.ipynb:136]:
```suggestion
    "Then, we load the source image, convert to gray scale and reshape it."
```

**Review by paularamo — CHANGES_REQUESTED** [review]
@FisherWY Thanks for your contributions. I have requested some changes to your notebook. Please update the Readme file, and include the links to the tools you are referring to.

@ryanloney after these changes, the notebook will be ready for your final review.

**Comment by FisherWY** [comment]
@paularamo Thank you for your suggestions, I have updated the notebook and README file according to your suggestions.

**Comment by raymondlo84** [comment]
@FisherWY -- Thank you again for this PR! :) We had reviewed it and we decided not to merge because it overlaps with our existing two superres ones already (e.g., https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/207-vision-paddlegan-superresolution)

=) Keep it up and we are looking forward to seeing more interesting notebooks from you next. 

P.S., not merging is not equal to fail the requirement.

**Comment by FisherWY** [comment]
> @FisherWY -- Thank you again for this PR! :) We had reviewed it and we decided not to merge because it overlaps with our existing two superres ones already (e.g., https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/207-vision-paddlegan-superresolution)
> 
> =) Keep it up and we are looking forward to seeing more interesting notebooks from you next.
> 
> P.S., not merging is not equal to failing the requirement.

Understood, although the pre-trained model used in this notebook has not been used, its task is still image super resolution.
Looking forward to launch more pre-trained models for different kinds of tasks. :)

**Pull request closed without merging**